modules/java/generator/gen_java.py

```diff
@@ -1347,7 +1347,7 @@ def gen_func(self, ci, fi, prop_name=''):
                 ret = "return (jlong) new %s(_retval_);" % self.fullTypeName(fi.ctype)
             elif fi.ctype.startswith('Ptr_'):
                 c_prologue.append("typedef Ptr<%s> %s;" % (self.fullTypeName(fi.ctype[4:]), fi.ctype))
-                ret = "return (jlong)(new %(ctype)s(_retval_));" % { 'ctype':fi.ctype }
+                ret = "%(ctype)s* curval = new %(ctype)s(_retval_);return (jlong)curval->get();" % { 'ctype':fi.ctype }
```

Comment:
@smigielski , looks like this change results in memory leak. The classes which wraps `Ptr<SomeObject>` stored address of `Ptr` object, but after this line has been changed, they store address of `SomeObject` itself. And several bytes from wrapping `Ptr` object will never be deleted. And object itself will be used and deleted incorrectly, because of being casted to `Ptr`.

Here is the example (see my comments with exclamation marks):
```.cpp
JNIEXPORT jlong JNICALL Java_org_opencv_ml_KNearest_create_10  (JNIEnv* env, jclass )
{
    static const char method_name[] = "ml::create_10()";
    try {
        LOGD("%s", method_name);
        typedef Ptr<cv::ml::KNearest> Ptr_KNearest;
        Ptr_KNearest _retval_ = cv::ml::KNearest::create(  );
        Ptr_KNearest* curval = new Ptr_KNearest(_retval_);return (jlong)curval->get(); // !!! return KNearest*
    } catch(const std::exception &e) {
        throwJavaException(env, &e, method_name);
    } catch (...) {
        throwJavaException(env, 0, method_name);
    }
    return 0;
}

JNIEXPORT void JNICALL Java_org_opencv_ml_KNearest_setEmax_10  (JNIEnv* env, jclass , jlong self, jint val)
{
    static const char method_name[] = "ml::setEmax_10()";
    try {
        LOGD("%s", method_name);
        Ptr<cv::ml::KNearest>* me = (Ptr<cv::ml::KNearest>*) self; //TODO: check for NULL // !!! cast to Ptr<KNearest>
        (*me)->setEmax( (int)val );
        return;
    } catch(const std::exception &e) {
        throwJavaException(env, &e, method_name);
    } catch (...) {
        throwJavaException(env, 0, method_name);
    }
    return;
}

JNIEXPORT void JNICALL Java_org_opencv_ml_KNearest_delete  (JNIEnv*, jclass, jlong self)
{
    delete (Ptr<cv::ml::KNearest>*) self; // !!! cast to Ptr<KNearest>
}
```